# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaDevices.cpp
# Commit: 9b14e7bdee7a
# Full Hash: 9b14e7bdee7a21cbd50f1cb0d57faa88a7025061
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-10-17 21:54:25
# Regressor Bug: 1397528
# File Overlap Count: 1
# Description:
#   Bug 1397528 - Handle cleaned up windows in async enumerateDevices response. r=jib
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D48661
# ==============================================================================

diff -r 58b4283727ea -r 9b14e7bdee7a dom/media/MediaDevices.cpp
--- a/dom/media/MediaDevices.cpp	Wed Oct 16 19:08:46 2019 +0000
+++ b/dom/media/MediaDevices.cpp	Thu Oct 17 07:57:13 2019 +0000
@@ -165,6 +165,7 @@
           },
           [this, self, p](const RefPtr<MediaMgrError>& error) {
             nsPIDOMWindowInner* window = GetWindowIfCurrent();
+            MOZ_DIAGNOSTIC_ASSERT(!window, "Should only fail after navigation");
             if (!window) {
               return;  // Leave Promise pending after navigation by design.
             }